src/components/material/infomedia/InfomediaModal.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import {
+  InfomediaService,
+  useGetInfomediaQuery
+} from "../../../core/dbc-gateway/generated/graphql";
+import { convertPostIdToFaustId } from "../../../core/utils/helpers/general";
+import Modal from "../../../core/utils/modal";
+import { useText } from "../../../core/utils/text";
+import { FaustId } from "../../../core/utils/types/ids";
+import InfomediaModalBody from "./InfomediaModalBody";
+import { Manifestation } from "../../../core/utils/types/entities";
+
+export const infomediaModalId = (faustId: FaustId) =>
+  `infomedia-modal-${faustId}`;
+
+interface InfomediaModalProps {
+  mainManifestation: Manifestation;
+}
+
+const InfomediaModal: React.FunctionComponent<InfomediaModalProps> = ({
+  mainManifestation,
+  mainManifestation: { pid }
+}) => {
+  const t = useText();
+
+  const infomediaAccess = mainManifestation.access.find(
+    (item) => item.__typename === "InfomediaService"
+  ) as InfomediaService;
+
+  const { data, error } = useGetInfomediaQuery({
+    id: infomediaAccess?.id
+  });
+
+  if (!data || error) {
+    return null;
+  }
+
+  const headline = data?.infomedia?.article?.headLine;
+  const text = data?.infomedia?.article?.text;
+
+  return (
+    <Modal
+      modalId={infomediaModalId(convertPostIdToFaustId(pid))}
```

Comment:
Consider changing `infomediaModalId()` to accept a `pid` instead of a `faust`.

It seems as if everywhere we use the function we convert a pid to a faust. Perhaps we can just let the function accept a pid instead? Then we can either convert it to a faust internally or perhaps work with it be as a pid.